presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -161,7 +162,26 @@ protected Boolean visitAtTimeZone(AtTimeZone node, Boolean isAggregation)
         @Override
         protected Boolean visitSubqueryExpression(SubqueryExpression node, Boolean isAggregation)
         {
-            return true;
+            return AstUtils.preOrder(node)
+                    .filter(columnReferences::containsKey)
+                    .map(Expression.class::cast)
+                    .allMatch(this::analyzeSubqueryReference);
+        }
+
+        private boolean analyzeSubqueryReference(Expression referenceExpression)
+        {
+            FieldId fieldId = requireNonNull(columnReferences.get(referenceExpression), "No FieldId for Expression");
+            if (!Objects.equals(fieldId.getRelationId(), sourceScope.getRelationType().getRelationId())) {
+                // Not our scope, not our responsibility
+                return true;
+            }
+
+            if (groupingFields.contains(fieldId)) {
```

Comment:
if a `fieldId` has an order by scope (`QuerySpecification` output column), then you should return true too.